rosidl_generator_type_description/resource/HashedTypeDescription.schema.json

```diff
@@ -0,0 +1,114 @@
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "$id": "HashedTypeDescription.schema.json",
+  "title": "HashedTypeDescription",
+  "description": "Contains hashes and full type description for a ROS 2 interface. TypeDescription, IndividualTypeDescription, Field, and FieldType are exact representations of type_description_interfaces/msg types, see their .msg files for semantic comments.",
+  "type": "object",
+  "properties": {
+    "hashes": {
+      "type": "object",
+      "oneOf": [
+        { "$ref": "#/$defs/MessageHash" },
+        { "$ref": "#/$defs/ServiceHashes" },
+        { "$ref": "#/$defs/ActionHashes" }
+      ]
+    },
+    "type_description": { "$ref": "#/$defs/TypeDescription" }
```

Comment:
Maybe it should have a different name, but I've got an infrastructure-defined JSON format here that is:
```
HashTree hashes;
type_description_interfaces/msg/TypeDescription type_description;
```

and `type_description_interfaces/msg/TypeDescription` is formatted like
```
IndividualTypeDescription type_description ;  // this type's description
IndividualTypeDescripiton[] referenced_type_descriptions;  // all the other ones
```

So that's where the nesting you're seeing is coming from. Perhaps it's a little more clear if the toplevel infrastructure format calls the field `type_description_msg`?

Reply:
> Perhaps it's a little more clear if the toplevel infrastructure format calls the field type_description_msg?

Yeah, I think that would be a little more clear to me.  Do you mind making that change?